src/components/ItemList.js

```diff
@@ -1,41 +1,67 @@
-import { collection, query, where, onSnapshot } from 'firebase/firestore';
-import { useEffect, useState } from 'react';
-import { db } from '../lib/firebase.js';
+import { useState } from 'react';
+import useFirebaseSnapshot from '../hooks/useFirebaseSnapshot.js';
+import cleanData from '../utils/cleanData.js';
 import { Link } from 'react-router-dom';
 
 const ItemList = () => {
-  const [docs, setDocs] = useState([]);
-
-  useEffect(() => {
-    const token = localStorage.getItem('token');
-    const q = query(
-      collection(db, 'shopping-list'),
-      where('token', '==', token),
-    );
-    const unsubscribe = onSnapshot(q, (querySnapshot) => {
-      const items = [];
-      querySnapshot.forEach((doc) => {
-        items.push(doc.data());
-      });
-      setDocs(items);
+  const [searchInput, setSearchInput] = useState('');
+  const [filteredResults, setFilteredResults] = useState('');
+
+  const { docs, loading } = useFirebaseSnapshot();
+
+  const filterItems = (query) => {
+    setSearchInput(query);
+    const results = docs.filter((item) => {
+      const name = cleanData(item.name);
+      return name.includes(cleanData(query));
     });
-    return () => {
-      unsubscribe();
-    };
-  }, []);
+    setFilteredResults(results);
+  };
+
+  const handleClear = () => {
+    setSearchInput('');
+    setFilteredResults('');
+  };
 
   return (
     <>
       <h2>Smart Shopping List</h2>
-      <ul>
-        {docs.length > 0 ? (
-          docs.map((item) => <li key={item.token + item.name}>{item.name}</li>)
-        ) : (
-          <p>
-            No items yet! <Link to="/add-item">Add one.</Link>
-          </p>
-        )}
-      </ul>
+      {loading && <p>Loading ...</p>}
+
+      {!docs.length && !loading && (
+        <p>
+          No items yet! <Link to="/add-item">Add some.</Link>
+        </p>
+      )}
+
+      {docs.length > 0 && (
+        <>
+          <form>
+            <label htmlFor="filter-items">Filter Items</label>
+            <input
+              id="filter-items"
+              type="text"
+              name="filter-items"
+              value={searchInput}
+              onChange={({ target }) => filterItems(target.value)}
+            />
+            <label htmlFor="btn">clear</label>
+            <input
+              type="btn"
+              id="btn"
+              className="btn-filter"
+              onClick={handleClear}
+            />
```

Comment:
This code caught my attention. It looks like the label itself is functioning as the clear button, being you have the input element hidden with css. However clicking on the label fires the handleClear function being its tied to the input? Do I have that correct? It seems like an interesting solution and I'm curious about this one! Nice!

Reply:
I'm curious to hear more about this approach as well. Is this a customary way of implementing this functionality?

Reply:
Initially, the button tag didn't have a type, was just calling the function, so page was reloading. Google gave me the input tag solution. Changing the tag back to button and having the type='button' solves the problem of page not reloading.